Spotify/module.php

```diff
@@ -424,6 +535,45 @@ public function SetShuffle(bool $Shuffle)
             $this->SetValue('Shuffle', $Shuffle);
         }
 
+        public function UpdatePosition()
+        {
+            if ($this->ReadAttributeString('Token') != '') {
+                $currentPlay = $this->requestCurrentPlay();
+                if (empty($currentPlay)) {
+                    $this->SetValue('Position', 0);
+                } else {
+                    $current_volume = $currentPlay['device']['volume_percent'];
+                    $this->SetValue('Volume', $current_volume);
+                    $this->SendDebug('Volume', $current_volume, 0);
+
+                    $progress_ms = $currentPlay['progress_ms'];
+                    $this->SendDebug('Progress ms', $progress_ms, 0);
+                    $this->SendDebug('Current Position ms', $progress_ms, 0);
+                    $progress = $this->ConvertTimeInterval($progress_ms);
```

Comment:
Da ConvertTimeInterval nur in dieser Funktion genutzt wird, definiere das gerne vor Ort als anonyme Funktion, siehe auch hier: https://www.php.net/manual/de/functions.anonymous.php